app/src/main/java/org/ole/planet/myplanet/MainApplication.kt

```
@@ -146,22 +146,30 @@ class MainApplication : Application(), WorkManagerConfiguration.Provider {
-                CoreDependenciesEntryPoint::class.java
-            )
-            val diagnosticsRepository = entryPoint.diagnosticsRepository()
-            return diagnosticsRepository.saveLogToRoom(type, error, time)
+            return try {
+                val entryPoint = EntryPointAccessors.fromApplication(
+                    context,
+                    CoreDependenciesEntryPoint::class.java
+                )
+                val diagnosticsRepository = entryPoint.diagnosticsRepository()
+                diagnosticsRepository.saveLogToRoom(type, error, time)
+            } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)